### PR TITLE
various: replace dead links in comments

### PR DESCRIPTION
**Pull request by na-na-hi**
Replace various dead links with live replacements or archives. Less friction for anyone who wants to look up these references.

### CODE REVIEW

File: osdep/subprocess-win.c [hunk 1]
```diff
@@ -402,6 +403,7 @@ void mp_subprocess2(struct mp_subprocess_opts *opts,
     // Specify which handles are inherited by the subprocess. If this isn't
     // specified, the subprocess inherits all inheritable handles, which could
     // include handles created by other threads. See:
+    // https://web.archive.org/web/20121127222200/
     // http://blogs.msdn.com/b/oldnewthing/archive/2011/12/16/10248328.aspx
```
Comment (kasper93) [inline osdep/subprocess-win.c:407]: Could link to the new place where this post reside https://devblogs.microsoft.com/oldnewthing/20111216-00/?p=8873

Comment (kasper93) [reply osdep/subprocess-win.c:407]: Agreed.

File: osdep/subprocess-win.c [hunk 2]
```diff
@@ -377,6 +377,7 @@ void mp_subprocess2(struct mp_subprocess_opts *opts,
     // Get pointers to the arrays in lpReserved2. This is an undocumented data
     // structure used by MSVCRT (and other frameworks and runtimes) to emulate
     // FD inheritance. The format is unofficially documented here:
+    // https://web.archive.org/web/20221014190010/
```
Comment (sfan5) [inline osdep/subprocess-win.c:380]: would be good to wrap in <> so they aren't mistaken for two links IMO

File: video/out/gpu/video_shaders.c [hunk 3]
```diff
@@ -254,7 +254,7 @@ void pass_compute_polar(struct gl_shader_cache *sc, struct scaler *scaler,
 static void bicubic_calcweights(struct gl_shader_cache *sc, const char *t, const char *s)
 {
     // Explanation of how bicubic scaling with only 4 texel fetches is done:
-    //   http://www.mate.tue.nl/mate/pdfs/10318.pdf
+    //   https://web.archive.org/web/20180720154854/http://www.mate.tue.nl/mate/pdfs/10318.pdf
```
Comment (sfan5) [inline video/out/gpu/video_shaders.c:257]: <>

**Review by sfan5 — APPROVED** [review]

**Pull request merged**